### PR TITLE
update micro-boot to Spring Boot with plugins:

### PR DESCRIPTION
**Pull request by johnmcclean**
micro-boot now runs as a Spring Boot app (supporting non-jersey front ends) with Microserver plugins (rest / web end points unavailable directly).

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: micro-boot/src/test/java/app/spring/plugin/RegistryTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    AsyncRestClient rest = new AsyncRestClient(1000,1000).withAccept("text/plain");
+
+    @Test
+    public void runAppAndBasicTest() throws InterruptedException, ExecutionException {
+
+        new MicroserverApp( ()-> "spring-mvc");
+        Thread.sleep(2000);
+
+//        assertThat(rest.get("http://localhost:8080/spring-mvc/sent").get(),is("0"));
```
Comment (morrowgi) [inline micro-boot/src/test/java/app/spring/plugin/RegistryTest.java]: Please Remove

File: micro-boot/src/main/java/com/oath/micro/server/boot/BootPlugin.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	@Override
+	public Set<Class> springClasses() {
+        return SetX.empty();
+    }
+	
+	@Override
+	public Function<FeatureContext,Map<String,Object>> jacksonFeatureProperties(){
+		return context-> new HashMap<>();
+	}
+
```
Comment (morrowgi) [inline micro-boot/src/main/java/com/oath/micro/server/boot/BootPlugin.java]: Remove extra spaces

File: micro-boot/src/main/java/com/oath/micro/server/rest/jersey/SpringBootJerseyRestApplication.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+		
+		context.getBeansOfType(AbstractBinder.class).forEach((n,e)->register(e));
+
+        JacksonJaxbJsonProvider p = new JacksonJaxbJsonProvider();
+        p.setMapper(JacksonUtil.getMapper());
+		register(p);
+		module.getDefaultJaxRsPackages().stream().forEach( e -> packages(e));
+		module.getDefaultResources().stream().forEach( e -> register(e));
+
```
Comment (morrowgi) [inline micro-boot/src/main/java/com/oath/micro/server/rest/jersey/SpringBootJerseyRestApplication.java]: Formatting

File: micro-application-register/src/main/java/com/oath/micro/server/application/registry/Job.java [hunk 4]
```diff
@@ -30,6 +31,12 @@
     private final String resourcePath;
     private final RegistryHealthChecker checker;
     private final RegistryStatsChecker statsChecker;
+    @Getter
+    private int scheduled =0;
+    @Getter
+    private int sent =0;
```
Comment (morrowgi) [inline micro-application-register/src/main/java/com/oath/micro/server/application/registry/Job.java]: I don't see this being used anywhere, as in no getSent() method call or geterrors()?

Comment (johnmcclean) [reply micro-application-register/src/main/java/com/oath/micro/server/application/registry/Job.java]: will remove

**Pull request merged**